libsolutil/Uri.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libsolutil/Uri.h>
+
+#include <array>
+#include <string>
+#include <string_view>
+
+using std::nullopt;
+using std::optional;
+using std::string;
+using std::string_view;
+
+namespace solidity::util
+{
+
+optional<Uri> Uri::parse(std::string_view _uri)
```

Comment:
Can't this be done by a single regex?

Reply:
Yes. I can use  `std::regex` for that.

Reply:
I was checking that now, but it seems, while the regex like can be all put into a single line, it'll be terribly unreadable and unmaintainable. At least for what I have found so far. The current code (as for file URIs) is tested, I am not sure regex will make that any easier. hmm....